### PR TITLE
[FR] Add ML Jobs to Schemas and Unit Test for Validation

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/security-team/issues/7421

## Summary
This PR adds the following:

- When building integration schemas, if the integration is an advanced analytic package, add ML job IDs to the final schema as `jobs`.
- Updates existing, `lmd` schema to include machine learning jobs for each version of the package
- Adds a unit test `test_ml_integration_jobs_exist` to ensure that any analytic package ML rule's `machine_learning_job_id` exists in the integrations schema. This happens by checking the min stack version or branch version, finding a compatible version from the integration manifests and job comparing job IDs.

## Screenshots
The following screenshots show the unit test failing with a rule where it sources from the LMD analytic package and has an unknown ML job ID that is not found in the LMD integration package version 1.0.2. Once fixed, the unit test passes.

<details>
Screenshot 2023-10-10 at 5 16 19 PM
Screenshot 2023-10-10 at 5 19 47 PM
</details>

Manifests and schemas were added for each analytic package in this PR.

<details>

```console
package: ded
 {'1.0.0': {'jobs': ['high-sent-bytes-destination-geo-city_name', 'high-sent-bytes-destination-geo-continent_name', 'high-sent-bytes-destination-geo-country_iso_code', 'high-sent-bytes-destination-geo-country_name', 'high-sent-bytes-destination-ip', 'high-sent-bytes-destination-port', 'high-sent-bytes-destination-region_name', 'high-sent-bytes-destination-timezone']}, '1.0.1': {'jobs': ['high-sent-bytes-destination-geo-city_name', 'high-sent-bytes-destination-geo-continent_name', 'high-sent-bytes-destination-geo-country_iso_code', 'high-sent-bytes-destination-geo-country_name', 'high-sent-bytes-destination-ip', 'high-sent-bytes-destination-port', 'high-sent-bytes-destination-region_name', 'high-sent-bytes-destination-timezone']}, '1.0.2': {'jobs': ['high-sent-bytes-destination-geo-city_name', 'high-sent-bytes-destination-geo-continent_name', 'high-sent-bytes-destination-geo-country_iso_code', 'high-sent-bytes-destination-geo-country_name', 'high-sent-bytes-destination-ip', 'high-sent-bytes-destination-port', 'high-sent-bytes-destination-region_name', 'high-sent-bytes-destination-timezone']}, '1.0.3': {'jobs': ['high-sent-bytes-destination-geo-country_iso_code', 'high-sent-bytes-destination-ip', 'high-sent-bytes-destination-port', 'high-sent-bytes-destination-region_name', 'high-bytes-written-to-external-device', 'rare-process-writing-to-external-device', 'high-bytes-written-to-external-device-airdrop']}}

package: lmd
 {'1.0.0': {'jobs': ['high-count-remote-file-transfer', 'high-file-size-remote-file-transfer', 'rare-file-extension-remote-transfer', 'rare-file-path-remote-transfer']}, '1.0.1': {'jobs': ['high-count-remote-file-transfer', 'high-file-size-remote-file-transfer', 'rare-file-extension-remote-transfer', 'rare-file-path-remote-transfer']}, '1.0.2': {'pivot_transform': {'host.name': 'keyword', 'user.name': 'keyword', 'process.Ext.authentication_id': 'keyword', 'destination.ip': 'ip', 'source.ip': 'ip', 'session.start_time': 'date', 'session.complete_time': 'date', 'number_processes_per_session': 'long', 'session.duration': 'float', 'total_length_process_args': 'double'}, 'jobs': ['high-count-remote-file-transfer', 'high-file-size-remote-file-transfer', 'rare-file-extension-remote-transfer', 'rare-file-path-remote-transfer', 'high-mean-rdp-session-duration', 'high-var-rdp-session-duration', 'high-sum-rdp-number-of-processes', 'unusual-time-weekday-rdp-session-start', 'high-rdp-distinct-count-source-ip-for-destination', 'high-rdp-distinct-count-destination-ip-for-source', 'high-mean-rdp-process-args']}}

package: problemchild
 {'1.0.0': {'jobs': ['problem_child_rare_process_by_host', 'problem_child_high_sum_by_host', 'problem_child_rare_process_by_user', 'problem_child_rare_process_by_parent', 'problem_child_high_sum_by_user', 'problem_child_high_sum_by_parent']}, '1.0.1': {'jobs': ['problem_child_rare_process_by_host', 'problem_child_high_sum_by_host', 'problem_child_rare_process_by_user', 'problem_child_rare_process_by_parent', 'problem_child_high_sum_by_user', 'problem_child_high_sum_by_parent']}, '1.1.0': {'jobs': ['problem_child_rare_process_by_host', 'problem_child_high_sum_by_host', 'problem_child_rare_process_by_user', 'problem_child_rare_process_by_parent', 'problem_child_high_sum_by_user', 'problem_child_high_sum_by_parent']}, '1.1.1': {'jobs': ['problem_child_rare_process_by_host', 'problem_child_high_sum_by_host', 'problem_child_rare_process_by_user', 'problem_child_rare_process_by_parent', 'problem_child_high_sum_by_user', 'problem_child_high_sum_by_parent']}, '1.1.2': {'jobs': ['problem_child_rare_process_by_host', 'problem_child_high_sum_by_host', 'problem_child_rare_process_by_user', 'problem_child_rare_process_by_parent', 'problem_child_high_sum_by_user', 'problem_child_high_sum_by_parent']}}

package: dga
 {'1.0.0': {'jobs': ['dga_high_sum_probability']}, '1.0.1': {'jobs': ['dga_high_sum_probability']}, '1.1.0': {'jobs': ['dga_high_sum_probability']}}
```

</details>

### CODE REVIEW

File: rules/ml/credential_access_ml_auth_spike_in_logon_events_from_a_source_ip.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ This rule uses a machine learning job to detect a substantial spike in successfu
 
 ### False positive analysis
 
-- Understand the context of the authentications by contacting the asset owners. If this activity is related to a new business process or newly implemented (approved) technology, consider adding exceptions — preferably with a combination of user and source conditions.
+- Understand the context of the auzthentications by contacting the asset owners. If this activity is related to a new business process or newly implemented (approved) technology, consider adding exceptions — preferably with a combination of user and source conditions.
```
Comment (terrancedejesus) [inline rules/ml/credential_access_ml_auth_spike_in_logon_events_from_a_source_ip.toml:44]:
```suggestion
- Understand the context of the authentications by contacting the asset owners. If this activity is related to a new business process or newly implemented (approved) technology, consider adding exceptions — preferably with a combination of user and source conditions.
```

File: tests/test_all_rules.py [hunk 2]
```diff
@@ -894,8 +894,39 @@ def test_all_min_stack_rules_have_comment(self):
...
+                        compat_integration_schema = integration_schemas[ml_integration_name][latest_compat_ver[0]]
+                        if rule_job_id not in compat_integration_schema['jobs']:
+                            failures.append(f'{self.rule_str(rule)} machine_learning_job_id `{rule_job_id}` not \
+                                            found in version `{latest_compat_ver[0]}` of `{ml_integration_name}` integration')
+        if failures:
+            err_msg = '\n'.join(failures)
+            self.fail(f'The following ({len(failures)}) rules are missing a valid `machine_learning_job_id`:\n{err_msg}')
 
-class TestRuleTiming(BaseRuleTest):
+class  TestRuleTiming(BaseRuleTest):
```
Comment (terrancedejesus) [inline tests/test_all_rules.py:929]:
```suggestion
class TestRuleTiming(BaseRuleTest):
```

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle:  LGTM, great walkthrough! :+1: 

Nit: Since you are updating the imports in `test_all_rules.py` you may want to run `Organize Imports`

organize_imports

**Comment by ajosh0504** [comment]
I see gzipped manifests and schema files being added in this PR- will these need to be refreshed if/when the manifests/schema is updated in the integrations repo?

**Comment by terrancedejesus** [comment]
@ajosh0504 

>I see gzipped manifests and schema files being added in this PR- will these need to be refreshed if/when the manifests/schema is updated in the integrations repo?

Yes. We will want to merge in the updates in the integration manifest and immediately after merge in the new rules and pull the manifests/schema updates.

**Review by imays11 — APPROVED** [review]
👍🏾

**Pull request merged**